pkg/storage/s3/client.go

```diff
@@ -118,16 +117,12 @@ func NewS3Client(kubeClient client.Client, region string) (Client, error) {
 	var err error
 
 	awsConfig := &aws.Config{Region: aws.String(region)}
-	namespace, err := k8sutil.GetOperatorNamespace()
-	if err != nil {
-		return nil, fmt.Errorf("failed to get operator namespace: %v", err)
-	}
 
 	secret := &corev1.Secret{}
 	err = kubeClient.Get(context.TODO(),
 		types.NamespacedName{
 			Name:      awsCredsSecretName,
-			Namespace: namespace,
+			Namespace: version.OperatorNamespace,
```

Comment:
We *shouldn't* hard code this. Getting the namespace via the downward API is important.

Reply:
`k8sutil.GetOperatorNamespace()` is hard-coded to read the namespace from `/var/run/secrets/kubernetes.io/serviceaccount/namespace` on whatever machine is running the code. For this patch specifically, that means that it's impossible to test this file without gaining root permissions, making the directory and writing a namespace to the file.

Unless I'm missing something, relying on `k8sutil.GetOperatorNamespace()` pretty much makes it impossible to unit test this function.

Reply:
Do both storage drivers use namespace? I think they do. I'd suggest that we pass `request.Namespace` down from the reconcile loop into the storage driver request here: https://github.com/openshift/managed-velero-operator/blob/3829d1b36db9e631666e90ae48162b3308cc5ee0/pkg/controller/velero/controller.go#L158, and then use that instead of `k8sutil.GetOperatorNamespace()` then